app/src/main/AndroidManifest.xml

```diff
@@ -5,6 +5,7 @@
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
```

Comment:
whats the reasoning for requiring network state here?

Reply:
Because I need to check whether the user is connected to the network or not in order to display messages or change the state of the app when the user is offline